Theory.txt

```diff
@@ -0,0 +1,14 @@
+What is the difference between IQueryable and IEnumerable interfaces? Provide some examples. 
+IEnumerable<T> interface consists of a set of extension methods like Where(), Select(), First(), Count(), 
+which ultimately gives the developer the ability to consistently write queries for collection of objects in memory (LINQ to Objects)
+To implement LINQ providers to data external to the application, the IQueryable<T> interface (an inheritor of IEnumerable<T>) is used along with 
+a set of extension methods that are almost identical to those written for IEnumerable<T>. IQueryable<T> interface inherits from IEnumerable<T>.
```

Comment:
The main difference is IEnumerable is in-memory collection while IQueryable just a query to external collection. When you use IEnumerable, the entire data set is loaded into memory and then filtered, sorted, or projected. This can be inefficient when working with large data sets. IQueryable, on the other hand, is designed to work with data sources that support querying, such as databases. When you use IQueryable, the query is not executed immediately. Instead, it is translated into a query that can be executed against the data source. This can be more efficient when working with large data sets.